### PR TITLE
`azurerm_netapp_volume_group_sap_hana` - update the validation of `security_style`

### PR DESCRIPTION
**Pull request by neil-yechenwei**
fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/22582

image

Note: Currently all related test cases are failed in Teamcity Daily Run due to quota issue. I found seems this resource is only available in eastus so that I updated the test location. However, its' pair region westus is still unavailable so that TestAccNetAppVolumeGroupSapHana_crossRegionReplication is still failed. But this failed test case is not related with this PR.

### CODE REVIEW

File: internal/services/netapp/validate/volume_group_sap_hana_volumes_validation.go [hunk 1]
```diff
@@ -67,8 +67,8 @@ func PossibleValuesForProtocolTypeVolumeGroupSapHana() []string {
 type SecurityStyle string
 
 const (
-	SecurityStyleUnix SecurityStyle = "Unix"
-	SecurityStyleNtfs SecurityStyle = "Ntfs"
+	SecurityStyleUnix SecurityStyle = "unix"
```
Comment (catriona-m) [inline internal/services/netapp/validate/volume_group_sap_hana_volumes_validation.go:70]: could we remove these and use the enum mentioned above instead?

Comment (neil-yechenwei) [reply internal/services/netapp/validate/volume_group_sap_hana_volumes_validation.go]: Updated to use enum values in go azure sdk.

**Comment by neil-yechenwei** [comment]
@catriona-m , thanks for the comment. I updated PR. Please take another look. Thanks.

File: internal/services/netapp/netapp_volume_group_sap_hana_resource.go [hunk 2]
```diff
@@ -164,11 +164,13 @@ func (r NetAppVolumeGroupSapHanaResource) Arguments() map[string]*pluginsdk.Sche
 						},
 					},
 
+					// Diverging from the SDK volumegroups.SecurityStyle since it is defined as lower case
+					// but the backend changes it to Pascal case on GET. Please refer to https://github.com/Azure/azure-sdk-for-go/issues/14684
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply internal/services/netapp/netapp_volume_group_sap_hana_resource.go]: Removed.

**Comment by neil-yechenwei** [comment]
@catriona-m , thanks for the comment. I updated PR. Please take another look. Thanks.

**Comment by neil-yechenwei** [comment]
@catriona-m , updating the validation may bring breaking change. Is it acceptable?

**Review by catriona-m — APPROVED** [review]
Thanks @neil-yechenwei LGTM!

**Pull request merged**